server/region_statistics.go

```diff
@@ -104,6 +104,58 @@ func (r *regionStatistics) Observe(region *core.RegionInfo, stores []*core.Store
 	}
 	r.deleteEntry(deleteIndex, regionID)
 	r.index[regionID] = peerTypeIndex
+	if len(stores) == 0 {
+		return
+	}
+	regionLabelLevel := getRegionLabelIsolationLevel(stores, labels)
+	regionLabelLevelHistogram.Observe(float64(regionLabelLevel))
+}
+
+func getRegionLabelIsolationLevel(stores []*core.StoreInfo, labels []string) int {
+	if len(stores) == 0 || len(labels) == 0 {
+		return -1
+	}
+	var queueStores [][]*core.StoreInfo
+	queueStores = append(queueStores, stores)
+	for level, label := range labels {
+		dequeNumber := len(queueStores)
+		isInLabelIsolation := true
+		for _, stores := range queueStores {
+			higherStores := higherLabelLeverStores(stores, label)
+			if len(higherStores) > 0 {
+				isInLabelIsolation = false
+				queueStores = append(queueStores, higherStores...)
+			}
+		}
+		queueStores = queueStores[dequeNumber:]
+		if isInLabelIsolation {
+			return level
+		}
+	}
+	return -1
+}
+
+func higherLabelLeverStores(stores []*core.StoreInfo, label string) [][]*core.StoreInfo {
```

Comment:
Lever -> Level. And the function name looks ambiguous, maybe we can figure out something better.